src/main/java/frc/robot/commands/CarouselCommand.java

```diff
@@ -35,14 +37,25 @@ public CarouselCommand(Carousel carousel) {
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
-    // every time the command is remade, the robot sets its state to waiting for ball
-    state = State.WaitingForBall;
+    // check the carousel is on the center of a slot
+    double slotError = Math.abs(Math.round(carousel.getSlot()) - carousel.getSlot());
+    if (slotError <= 0.1){
+      // carousel is aligned, wait for ball
+      state = State.WaitingForBall;
+    }
+    else{
+      // carousel not aligned, rotate to the next slot
+      moveToNextSlot();
+    }
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
   
+    // Print state value to smart dashboard
+    SmartDashboard.putString("carousel/state", state.toString());
```

Comment:
this ought to be "Carousel" to match what is in Carousel.

Reply:
Agree